From bf7c05d0366cd8b85ce9d74341ed6560b6dfbd65 Mon Sep 17 00:00:00 2001
From: Steve Langasek <steve.langasek@canonical.com>
Date: Wed, 28 Sep 2011 12:37:42 -0700
Subject: [PATCH] pass dpkg-buildflags to makemaker build system

The standard way to pass build flags to makemaker perl build systems is to
set the OPTIMIZE variable on the commandline; CFLAGS in the environment gets
ignored entirely.  So pass the CFLAGS from the environment to Makefile.PL so
makemaker packages can also benefit from dpkg-buildflags.
---
 Debian/Debhelper/Buildsystem/perl_makemaker.pm |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/Debian/Debhelper/Buildsystem/perl_makemaker.pm b/Debian/Debhelper/Buildsystem/perl_makemaker.pm
index 72f8d8c..0e9ff4c 100644
--- a/Debian/Debhelper/Buildsystem/perl_makemaker.pm
+++ b/Debian/Debhelper/Buildsystem/perl_makemaker.pm
@@ -39,17 +39,22 @@ sub new {
 
 sub configure {
 	my $this=shift;
+	my @flags;
 	# If set to a true value then MakeMaker's prompt function will
 	# # always return the default without waiting for user input.
 	$ENV{PERL_MM_USE_DEFAULT}=1;
 	# This prevents  Module::Install from interactive behavior.
 	$ENV{PERL_AUTOINSTALL}="--skipdeps";
 
+        if ($ENV{CFLAGS}) {
+                push @flags, "OPTIMIZE=$ENV{CFLAGS}";
+        }
+
 	$this->doit_in_sourcedir("perl", "Makefile.PL", "INSTALLDIRS=vendor",
 		# if perl_build is not tested first, need to pass packlist
 		# option to handle fallthrough case
 		(compat(7) ? "create_packlist=0" : ()),
-		@_);
+		@flags, @_);
 }
 
 sub install {
-- 
1.7.5.4

